src/api/c/implicit.cpp

```diff
@@ -47,6 +47,12 @@ af_dtype implicit(const af_dtype lty, const af_dtype rty)
     if ((lty == s32) ||
         (rty == s32)) return s32;
 
+    if ((lty == u16) ||
+        (rty == u16)) return u16;
+
+    if ((lty == s16) ||
+        (rty == s16)) return s16;
+
```

Comment:
I did not think of this until I actually saw the change. Can you look at src/api/cpp and ensure there is no implicit.cpp in there as well ?